src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6441,6 +6459,20 @@ bool CracRestoreParameters::read_from(int fd) {
     cursor = cursor + prop_len;
   }
 
+  {
+    char* env_mem = NEW_C_HEAP_ARRAY(char, hdr->_env_memory_size, mtArguments); // left this pointer unowned, it is freed when process dies
```

Comment:
```suggestion
  char* env_mem = NEW_C_HEAP_ARRAY(char, hdr->_env_memory_size, mtArguments); // left this pointer unowned, it is freed when process dies
```

Here and below, we don't hesitate to introduce few local variables without creating a new syntax block. Please note the change once applied makes the code uncompilable.